src/JsonService.ts

```diff
@@ -131,12 +133,21 @@ export class JsonService {
         basicAuth,
         timeoutInSeconds,
         initCredentials,
+        dpopEnabled,
     }: PostFormOpts): Promise<Record<string, unknown>> {
         const logger = this._logger.create("postForm");
+        let DPoPProof;
+
         const headers: HeadersInit = {
             "Accept": this._contentTypes.join(", "),
             "Content-Type": "application/x-www-form-urlencoded",
         };
+
```

Comment:
instead of limiting this to DPoP header, we could simply enrich the interface with a generic: `extraHeaders?: Record<string, ExtraHeader>;` parameter...